src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmailController.php

```diff
@@ -0,0 +1,172 @@
+
+        $configuration = $this->get('prestashop.adapter.legacy.configuration');
+        $isEmailLogsEnabled = $configuration->get('PS_LOG_EMAILS');
+
+        $presentedEmailLogsGrid = null;
+
+        if ($isEmailLogsEnabled) {
+            $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+            $emailLogsGridFactory = $this->get('prestashop.core.grid.factory.email_logs');
+            $emailLogsGrid = $emailLogsGridFactory->createUsingSearchCriteria($filters);
+            $presentedEmailLogsGrid = $gridPresenter->present($emailLogsGrid);
+        }
+
+        return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/Email/email.html.twig', [
+            'emailConfigurationForm' => $emailConfigurationForm->createView(),
+            'isOpenSslExtensionLoaded' => $extensionChecker->loaded('openssl'),
+            'smtpMailMethod' => MailMethodOption::SMTP,
+            'testEmailSendingForm' => $testEmailSendingForm->createView(),
+            'emailLogsGrid' => $presentedEmailLogsGrid,
+            'isEmailLogsEnabled' => $isEmailLogsEnabled,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * Process email configuration saving
+     *
+     * @DemoRestricted(redirectRoute="admin_email")
+     * @AdminSecurity("is_granted(['update', 'create', 'delete'], request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processFormAction(Request $request)
+    {
+        $formHandler = $this->getEmailConfigurationFormHandler();
+        $emailConfigurationForm = $formHandler->getForm();
+        $emailConfigurationForm->handleRequest($request);
+
+        if ($emailConfigurationForm->isSubmitted()) {
+            $errors = $formHandler->save($emailConfigurationForm->getData());
+
+            if (empty($errors)) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+            } else {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_email');
+    }
+
+    /**
+     * Processes test email sending
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function processTestEmailSendingAction(Request $request)
+    {
+        if (!$request->isXmlHttpRequest()) {
+            return $this->redirectToRoute('admin_email');
+        }
+
+        if ($this->isDemoModeEnabled()) {
+            return $this->json([
+                'errors' => [
+                    $this->getDemoModeErrorMessage(),
+                ],
+            ]);
+        }
+
+        if (!in_array(
+            $this->authorizationLevel($request->attributes->get('_legacy_controller')),
```

Comment:
Use `@DemoRestricted` and `@AdminSecurity`

Reply:
not sure if i can do that, because it always sets `RedirectResponse` even if you make ajax call. :/

Reply:
Sure you can, even if it's a redirect response, it's executed before returning any response =) (or you can try :P)

Reply:
Use routing condition instead of check about ajax action in controller

Reply:
We may improve the annotation behavior according to the request type (ajax/not ajax/json or html expected). For now, I'm in favor of not relying on annotations here to keep the scope of this pr manageable, don't you think?